### ISSUE TITLE
Paths do not display legends

### ISSUE THREAD

**Issue by ea42gh** [ioam/holoviews#2601]
```
hv.Path( [ [[1,4],[4,1]],   [[3,3],[2,2]]], label='foo' )*hv.Path( [ [[4,4],[1,1]],   [[3,2],[2,3]]], label='bah' )
```
does not produce legends

**Comment by ea42gh** [comment]
Found the option:  show_legend=True
To be consistent with other elements, should this be the default?

**Comment by jlstevens** [comment]
It seems there are some element types where you want legends on my default and others where you don't (e.g Curves versus Paths). If we decide to make such a distinction, what we need to do is document it and list the elements based on their defaults.

**Comment by ea42gh** [comment]
Not sure why you say that? I have to explicitly add a `label=...`  to the call to get the legend.
So with paths, this becomes `hv.Path(...,label='tata').options( show_legend=True)`.

**Comment by jlstevens** [comment]
It comes down to a judgment about how ``Paths`` are used. Whenever I have used this element myself, I know  I would always be switching the legend off. Of course, that doesn't mean it reflects everybody's experience so it is a value judgement about sensible default behavior.

**Comment by ea42gh** [comment]
But that is the question I was trying to raise: if you know you do not want the paths to have a legend,
would you not define
`labeled_elements * hv.Path( data )`  as opposed to
`labeled_elements * hv.Path( data, label='path')` ?

**Comment by jbednar** [comment]
Right; that's the question: @jlstevens, it's true that there are many uses of paths where a legend is not appropriate.  But are there a lot of uses for Paths *with labels* where a legend is not appropriate?  I agree with @ea42gh that a typical user would simply not declare the label in that case, and thus it seems to me that show_legend ought to be true here both to support that and to make things more consistent and predictable for users.

**Comment by philippjfr** [comment]
I agree consistency is preferable here, especially since you still have to define a ``label`` to get a legend.

**Comment by jlstevens** [comment]
I suppose when labels are defined you should have the same default for showing the legend.

I think I got confused thinking of ``Contours``, surely you can show the legend for the contour level values within a single ``Contours`` element without specifying a label? Or is this not possible?

**Comment by philippjfr** [comment]
The value judgement is indeed more relevant to Contours/Polygons here, adding a ``label`` based legend isn't controversial I don't think. The problem is that Contours/Polygons support adding legends for each unique contour/polygon generated from the value dimension that is being colormapped in which case enabling the legend by default can be pretty horrible.

**Comment by ea42gh** [comment]
Just got bit by this again with polygons today. Actually raised a question on gitter since I had forgotton about this behavior... I still think that an explicit `Element(...,label='tata')` statement should have show_legend=True by default

**Comment by ea42gh** [comment]
Seems I get bit by this once a year! Spent a couple of hours on this today yet again :-(
